lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3605,9 +3605,16 @@ public function mapEmbedded(array $mapping)
     {
         $this->assertFieldNotMapped($mapping['fieldName']);
 
+        if (! isset($mapping['class']) && $this->isTypedProperty($mapping['fieldName'])) {
+            $type = $this->reflClass->getProperty($mapping['fieldName'])->getType();
+            if ($type instanceof ReflectionNamedType) {
+                $mapping['class'] = $type->getName();
+            }
+        }
+
```

Comment:
You need to check here again that class isset or otherwise throw an exception. If not, then line 3616 will throw a notice accessing mapping class that does not exist.

Reply:
Actually the validation is done "late" in `ClassMetadataFactory::doLoadMetadata` as per https://github.com/doctrine/orm/pull/1133 - this should probably be changed, but not now.